scheduler/src/test/scala/com/sky/kms/base/KafkaIntSpecBase.scala

```diff
@@ -4,26 +4,35 @@ import com.sky.kms.kafka.Topic
 import com.sky.kms.utils.RandomPort.randomPort
 import eu.timepit.refined.auto._
 import net.manub.embeddedkafka.Codecs.{nullDeserializer, stringDeserializer}
-import net.manub.embeddedkafka.ConsumerExtensions.{ConsumerRetryConfig, _}
-import net.manub.embeddedkafka.{Consumers, EmbeddedKafka, EmbeddedKafkaConfig}
+import net.manub.embeddedkafka.{EmbeddedKafka, EmbeddedKafkaConfig}
 import org.apache.kafka.clients.consumer.{ConsumerRecord, KafkaConsumer}
 import org.apache.kafka.common.serialization.Deserializer
 import org.scalatest.WordSpecLike
 
-trait KafkaIntSpecBase extends EmbeddedKafka with WordSpecLike with Consumers {
+import collection.JavaConverters._
+import java.time.{Duration => JDuration}
+import scala.concurrent.duration._
+import scala.compat.java8.DurationConverters._
+
+trait KafkaIntSpecBase extends EmbeddedKafka with WordSpecLike {
 
   implicit lazy val kafkaConfig = EmbeddedKafkaConfig(kafkaPort = randomPort(), zooKeeperPort = randomPort())
 
   val scheduleTopic: Topic      = "scheduleTopic"
   val extraScheduleTopic: Topic = "extraScheduleTopic"
 
-  val retryConfig = ConsumerRetryConfig(maximumAttempts = 50)
-
-  def consumeFirstFrom[T : Deserializer](topic: String): ConsumerRecord[Array[Byte], T] =
-    withConsumer[Array[Byte], T, ConsumerRecord[Array[Byte], T]](_.consumeLazily(topic)(identity, retryConfig).head)
+  def consumeFirstFrom[T : Deserializer](topic: String,
+                                         timeout: FiniteDuration = 60.seconds): ConsumerRecord[Array[Byte], T] =
+    withConsumer[Array[Byte], T, ConsumerRecord[Array[Byte], T]] { cr =>
+      cr.subscribe(List(topic).asJavaCollection)
+      cr.poll(timeout.toJava).iterator().next()
+    }
 
-  def consumeSomeFrom[T : Deserializer](topic: String, numMsgs: Int): List[ConsumerRecord[String, T]] =
+  def consumeSomeFrom[T : Deserializer](topic: String,
+                                        numMsgs: Int,
+                                        timeout: FiniteDuration = 60.seconds): List[ConsumerRecord[String, T]] =
     withConsumer { cr: KafkaConsumer[String, T] =>
-      cr.consumeLazily(topic)(identity, retryConfig).take(numMsgs).toList
+      cr.subscribe(List(topic).asJavaCollection)
+      cr.poll(JDuration.ofSeconds(timeout.toSeconds)).iterator().asScala.toList.take(numMsgs)
```

Comment:
```suggestion
      cr.poll(timeout.toJava).iterator().asScala.toList.take(numMsgs)
```

Then the java import can be removed

Reply:
oops, missed that one, one before is doing exactly that. thx